server/client_server.go

```diff
@@ -22,21 +23,126 @@ func newServerClient(jiraClient *jira.Client) Client {
 	}
 }
 
+type IssueInfo struct {
+	Values []*jira.MetaIssueType `json:"values,omitempty"`
+}
+
+type FieldInfo struct {
+	Values []interface{} `json:"values,omitempty"`
+}
+
+type FieldValues struct {
+	Id string `json:"fieldId,omitempty"`
+}
+
+type FieldId struct {
+	Values []FieldValues `json:"values,omitempty"`
+}
+type Version struct {
+	VersionInfo string `json:"version,omitempty"`
+}
+
 // GetCreateMeta returns the metadata needed to implement the UI and validation of
 // creating new Jira issues.
 func (client jiraServerClient) GetCreateMeta(options *jira.GetQueryOptions) (*jira.CreateMetaInfo, error) {
-	cimd, resp, err := client.Jira.Issue.GetCreateMetaWithOptions(options)
+	apiEndpoint := "rest/api/2/serverInfo"
+	v := new(Version)
+	req, err := client.Jira.NewRequest("GET", apiEndpoint, nil)
+	if err != nil {
+		return nil, err
+	}
+	_, err = client.Jira.Do(req, v)
+	if err != nil {
+		return nil, err
+	}
+
+	v1, err := version.NewVersion(v.VersionInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	v2, err := version.NewVersion("8.4.0")
+	if err != nil {
+		return nil, err
+	}
+
+	var info *jira.CreateMetaInfo
+	var response *jira.Response
+	if v1.LessThan(v2) {
+		cimd, resp, error := client.Jira.Issue.GetCreateMetaWithOptions(options)
+		info = cimd
+		response = resp
+		err = error
+	} else {
+		cd, resp, err := client.Jira.Project.ListWithOptions(options)
+		meta := new(jira.CreateMetaInfo)
+
+		if err == nil {
+			for i := 0; i < len(*cd); i++ {
+				meta.Expand = (*cd)[i].Expand
+				project := new(jira.MetaProject)
```

Comment:
see if you can initialise the project directly with all the values you have updated individually from line 84 to 87